packages/nimble-components/src/wafer-map/modules/event-handler.ts

```diff
@@ -0,0 +1,49 @@
+import type { WaferMap } from '..';
+import type { WaferMapDie, EventHandlerData } from '../types';
+import { ZoomHandler } from './zoom-handler';
+
+/**
+ * EventHandler deals with user interactions and events
+ */
+export class EventHandler {
+    private readonly zoomHandler: ZoomHandler;
+
+    public constructor(private readonly eventHandlerData: EventHandlerData) {
+        this.zoomHandler = new ZoomHandler(eventHandlerData.zoomHandlerData);
+
+        // TODO HoverHandler - initialization
+
+        // TODO HoverHandler -  create a hoverDie element
+    }
+
+    public attachEvents(waferMap: WaferMap): void {
```

Comment:
Instead of a public API to attach events, can this work happen in the constructor?

Reply:
I think that it could and I did consider this, it's just that in this case we would need to modify the constructor to get also a reference to the `wafermap`, or to include it in the `EventHandlerData` ... Either way, by having it with this granularity level, it felt like the code was a little bit easier to follow ... I did not see a big enough gain in doing so at the moment.

Reply:
The advantage would be to simplify the public API of each of the "handler" classes. If you can pass everything you need in the constructor then it's one less API to worry about. It saves us from asking questions like "what if `attachEvents` is called twice?" or "why isn't there a `detachEvents`?" or "what if you forget to call `attachEvents`?"

I still think it would be an improvement. I believe it would be pretty simple to include the wafer map as part of `EventHandlerData`.

Reply:
ok ... applied